### DISCUSSION TITLE
Open the left panel widget in the main shell area.

### DISCUSSION THREAD

**Discussion by YathindraNikshipth** [eclipse-theia/theia#14546] [General] [answered]
I would like to Open the left panel widget in the main shell area, rather than rendering the content in the sidebar when it is clicked.is there a way to achieve this?

**Comment by msujew** [comment, marked as answer]
Hey @YathindraNikshipth,

I don't think there's a recommended way of doing this. The issue is that the sidebar is essentially a tabbar for the side (left/right), like the tabbar for editors in the main area. Meaning that any widget that resides on that sidebar tabbar really lives in the sidebar.

You might be able to make this work by replacing the tabbar renderer with a custom renderer that allows you to contribute a command instead of a widget. This command would then open the respective widget in the main area.

**Comment by YathindraNikshipth** [comment]
Hi @msujew, thanks for your reply. Could you please share a reference for creating a custom tab bar rendering?

**Reply by msujew** [reply]
There is no reference for such, since it is a very non-standard requirement. You will need to override the [SidebarTopMenuWidget](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/shell/sidebar-top-menu-widget.tsx). See also the associated [SidebarMenuWidget](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/browser/shell/sidebar-menu-widget.tsx).

**Reply by YathindraNikshipth** [reply]
Thank you, will look into that :)

**Reply by YathindraNikshipth** [reply]
Hey @msujew ,

I came up with a workaround where I created two widgets. In one of them, I override onActivationRequest to add the widget to the main panel and then collapse the left panel.

Do you think there could be any problems with this approach or anything I should watch out for?

Thanks

**Reply by msujew** [reply]
I think that's a reasonable solution. You might run into some minor unexpected issues, like the view appearing two times in the `View` menu (one for the left panel, and then one for the main panel), but I think the general approach is good 👍

**Reply by YathindraNikshipth** [reply]
@msujew That's right, I think I can filter that menu out. Thanks for your input.

**Reply by YathindraNikshipth** [reply]
@msujew , I found there might be one more issue. When the widget is in the additional views, clicking on it causes the left panel container to expand, but it seems that the onActivationRequest is not being triggered. Do you think something else could be responsible for this behavior?

**Reply by msujew** [reply]
Unfortunately, no idea. You might need to debug the lumino code for this.

**Reply by YathindraNikshipth** [reply]
@msujewI tried using lifecycle hooks in Lumino other than onActivationRequest, such as onBeforeShow. It allowed me to prevent the left panel from expanding, which is great. However, I noticed something odd — even when I click on the Explorer or any other widget on left panel, the focus remains on my widget. As a result, my widget’s functionality still gets triggered, and a new widget is placed in the main panel.

Could this be a bug or a build issue, or am I possibly missing something?

Could you please provide any pointers or clues that might help me understand this better?

Thank u.

`

     protected async onActivateRequest(msg: Message): Promise<void> {
        super.onActivateRequest(msg);

        this.shell.leftPanelHandler.collapse();

        await this.attachCustomWidget();

     }

    protected onBeforeShow(msg: Message): void {
        this.activate();
    }
`

**Reply by msujew** [reply]
I'm not that deep into the Lumino stack, maybe @sdirix can help you along.

**Reply by sdirix** [reply]
I'm not familiar enough with Lumino myself to explain what is going wrong. In general it seems like a rather "hacky" approach.

A clean approach I have seen in adopter projects is to subclass the application shell, replacing the whole left/right panels with own implementations. There you can implement arbitrary behavior. This will work but it's non-trivial.

**Comment by YathindraNikshipth** [comment]
@msujew @sdirix Thanks for the inputs, will try working on that!
